client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -193,12 +193,9 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 						key={ step.slug }
 						path={ `/${ flow.variantSlug ?? flow.name }/${ step.slug }` }
 						element={
-							<StepRoute
-								step={ step }
-								flow={ flow }
-								showWooLogo={ isWooExpressFlow( flow.name ) }
-								renderStep={ renderStep }
-							/>
+							<StepRoute step={ step } flow={ flow } showWooLogo={ isWooExpressFlow( flow.name ) }>
+								{ renderStep( step ) }
```

Comment:
This refactor feels like it may have a subtle side-effect: we may be changing when the rendering occurs. At present, we're passing the `renderStep` function into `StepRoute` and the function is called when the `StepRoute` is rendered and we're relying on closures to reach into the internal state of `FlowRenderer`. With this change, we're now rendering the content for all steps as we're rendering the main flow content. I can't tell whether that timing change will have any logical impacts, nor am I clear on whether there is any noticeable performance impact from pulling that rendering for all steps forward.

Reply:
I am going to rethink this change.